packages/serverpod/lib/src/server/server.dart

```diff
@@ -425,7 +436,8 @@ class Server {
     var path = uri.pathSegments.join('/');
     var endpointComponents = path.split('.');
     if (endpointComponents.isEmpty || endpointComponents.length > 2) {
-      return ResultInvalidParams('Endpoint $path is not a valid endpoint name');
+      return ResultNoSuchEndpoint(
```

Comment:
Suggestion: Revert this change.

As far as I understand, this is more of a formatting exception/invalid params exception rather than a no such endpoint exception right? And how does this tie together with the error validation done on line 472?

Reply:
Hampus made some effort consolidating our endpoint parsing logic so that it is a shared path regardless of how the protocol layer is implemented. So endpoint dispatch currently owns the knowledge about possible endpoints.

I think it is enough returning this exception based on that knowledge, as you have already implemented on line 538.